### PR TITLE
DE4163 - Add redirect functionality

### PR DESCRIPTION
**Pull request by chris-branch**

### CODE REVIEW

File: apps/crossroads_content/lib/content_pages.ex [hunk 1]
```diff
@@ -103,37 +103,127 @@ defmodule CrossroadsContent.Pages do
...
     CrossroadsContent.CmsClient.get("Page", Map.new |> set_angular_not_required |> set_stage(stage))
   end
 
-  defp get_non_angular_page(url, stage) do  
+  defp get_non_angular_page(url, stage) do
     CrossroadsContent.CmsClient.get("Page", Map.new |> set_angular_not_required |> set_stage(stage) |> set_link(url))
   end
 
+  defp get_redirector_pages(stage) do
+    redirector_pages = CrossroadsContent.CmsClient.get("Page", Map.new |> set_redirector_pagetype |> set_stage(stage))
```
Comment (mjvcallibrity) [inline apps/crossroads_content/lib/content_pages.ex]: Though nice to have on a single line, it might make reading easier to set the result of `Map.new |> set_redirector_pagetype |> set_stage(stage)` to a descriptive variable and pass that variable in as an argument to `CmsClient.get`.

File: apps/crossroads_content/lib/content_pages.ex [hunk 2]
```diff
@@ -103,37 +103,127 @@ defmodule CrossroadsContent.Pages do
...
+
+      cond do
+        url != nil -> Map.put(acc, page["link"], %{ "id" => page["id"], "link" => page["link"], "redirectUrl" => url })
+        true -> acc
+      end
+    end )
+  end
+
+  defp get_redirector_targets(redirector_pages) do
+    get_link_to = fn page ->
```
Comment (mjvcallibrity) [inline apps/crossroads_content/lib/content_pages.ex]: This particular function feels very _un_elixir-y. Ideally, this function would start with the `redirector_pages` variable and transform it through a serious of well-named private functions using the `|>` operator. That way it is more clear what transformations are taking place in this function.

File: apps/crossroads_content/lib/content_pages.ex [hunk 3]
```diff
@@ -103,37 +103,127 @@ defmodule CrossroadsContent.Pages do
...
+        "title" => page["title"],
+        "link" => page["link"],
+      }
+      Map.put(acc, page["id"], redirectData)
+    end )
+  end
+
+  defp get_id_chunks(ids) when length(ids) === 0 do [] end
+  defp get_id_chunks(ids) do
+    max_ids_per_call = 125;
```
Comment (mjvcallibrity) [inline apps/crossroads_content/lib/content_pages.ex]: Would this be an appropriate place to use Enumerable.chunk_every/4?

File: apps/crossroads_content/lib/content_pages.ex [hunk 4]
```diff
@@ -103,37 +103,131 @@ defmodule CrossroadsContent.Pages do
...
-  defp get_non_angular_page(url, stage) do  
-    CrossroadsContent.CmsClient.get("Page", Map.new |> set_angular_not_required |> set_stage(stage) |> set_link(url))
+  defp get_redirector_pages(stage) do
+    params = Map.new |> set_redirector_pagetype |> set_stage(stage)
+    redirector_pages = CrossroadsContent.CmsClient.get("Page", params)
+
+    case redirector_pages do
+      {:ok, 200, response} ->
+          id_map = get_redirector_targets(response)
+          {:ok, 200, get_redirects(response, id_map)}
```
Comment (richardbarker) [inline apps/crossroads_content/lib/content_pages.ex]: I'm finding this code hard to follow.  Can we chat about this?
